src/bin/ayb.rs

```diff
@@ -360,26 +374,58 @@ async fn main() -> std::io::Result<()> {
                 }
             }
         } else if let Some(matches) = matches.subcommand_matches("query") {
-            if let (Some(entity_database), Some(query), Some(format)) = (
+            if let (Some(entity_database), Some(format)) = (
                 matches.get_one::<EntityDatabasePath>("database"),
-                matches.get_one::<String>("query"),
                 matches.get_one::<OutputFormat>("format"),
             ) {
-                match client
+                if let Some(query) = matches.get_one::<String>("query") {
+                    match client
                     .query(&entity_database.entity, &entity_database.database, query)
                     .await
-                {
-                    Ok(query_result) => {
-                        if !query_result.rows.is_empty() {
-                            match format {
-                                OutputFormat::Table => query_result.generate_table()?,
-                                OutputFormat::Csv => query_result.generate_csv()?,
+                    {
+                        Ok(query_result) => {
+                            display_query_result(&query_result, format)?;
+                        }
+                        Err(err) => {
+                            println!("Error: {}", err);
+                        }
+                    }   
```

Comment:
I like that you created a function `display_query_result` since its logic is used in two places. There's an opportunity to expand that idea: It looks like this entire block (line 382-392) is the same as the one on 407-417. Instead of `display_query_result`, you could have a function `query_and_display(client, entity, database, query, format)` that covers both blocks.

Reply:
Yeah it would be better to avoid repeating code. Committed the change!